### PR TITLE
feat(imageResolver): add config options for image resolver

### PR DESCRIPTION
**Pull request by tonymbugua-outreach**
## Jira ID

DT-3506

### CODE REVIEW

**Review by kaldorn — APPROVED** [review]
That adds some structs.

File: pkg/box/box.go [hunk 1]
```diff
@@ -90,6 +92,34 @@ type VaultConfig struct {
 	AddressCI string `yaml:"addressCI"`
 }
 
+// VersionResolvers is the configurations used to get the latest version
+type VersionResolvers struct {
```
Comment (jaredallard) [inline pkg/box/box.go:96]: Might want to use `DevenvVersionResolvers` in the name, until we make this be used other places.

File: pkg/box/box.go [hunk 2]
```diff
@@ -90,6 +92,34 @@ type VaultConfig struct {
...
+	// enabled is a list of image resolvers to use. If none are specified Maestro will be used
```
Comment (jaredallard) [inline pkg/box/box.go:97]:
```suggestion
	// Enabled is a list of image resolvers to use. If none are specified Maestro will be used
```

File: pkg/box/box.go [hunk 3]
```diff
@@ -90,6 +92,34 @@ type VaultConfig struct {
...
+	// ordered based on priority
+	Enabled []string `yaml:"enabled"`
+
+	// Config is the configuration information for all versionResolvers
+	Config VersionResolverConfig `yaml:"config"`
+}
+
+// VersionResolverConfig list of versionResolver configurations
```
Comment (jaredallard) [inline pkg/box/box.go:105]:
```suggestion
// VersionResolverConfig contains configuration for version resolvers
```

File: pkg/box/box.go [hunk 4]
```diff
@@ -90,6 +92,34 @@ type VaultConfig struct {
...
+type VersionResolverConfig struct {
+	// Maestro configuration used by maestro image resolver
+	Maestro MaestroConfig `yaml:"maestro"`
+}
+
+// MaestroConfig configuration used by the maestro imageResolver
```
Comment (jaredallard) [inline pkg/box/box.go:111]:
```suggestion
// MaestroConfig contains configuration used by the maestro version resolver
```

File: pkg/box/box.go [hunk 5]
```diff
@@ -58,6 +58,8 @@ type DeveloperEnvironmentConfig struct {
 	// RuntimeConfig stores configuration specific to different devenv
 	// runtimes.
 	RuntimeConfig DeveloperEnvironmentRuntimeConfig `yaml:"runtimeConfig"`
+
+	VersionResolvers DevenvVersionResolvers `yaml:"versionResolvers"`
```
Comment (jaredallard) [inline pkg/box/box.go:62]: Comment for `VersionResolvers` since it's exported pls!

File: pkg/box/box.go [hunk 6]
```diff
@@ -90,6 +92,34 @@ type VaultConfig struct {
 	AddressCI string `yaml:"addressCI"`
 }
 
+// DevenvVersionResolvers is the configurations used to get the latest version
+type DevenvVersionResolvers struct {
+	// Enabled is a list of image resolvers to use. If none are specified Maestro will be used
+	// ordered based on priority
```
Comment (jaredallard) [inline pkg/box/box.go:98]: For external consumers, `git` should be the default, so I'd document that

File: pkg/box/box.go [hunk 7]
```diff
@@ -90,6 +92,34 @@ type VaultConfig struct {
...
+	Enabled []string `yaml:"enabled"`
+
+	// Config is the configuration information for all versionResolvers
```
Comment (jaredallard) [inline pkg/box/box.go:101]:
```suggestion
	// Config is the configuration information for all version resolvers
```

**Review by jaredallard — APPROVED** [review]

**Pull request merged**